### DISCUSSION TITLE
Calculate the next day when the star rises, whilst it's dark

### DISCUSSION THREAD

**Discussion by marcmayo** [cosinekitty/astronomy#303] [General]
Let's say we want to observe a star or a constellation (constellation would have min and max RA/DEC from the IAU constellation boundaries )
Can we somehow iterate through the next 365 days and tell at which day the star finally rises above the horizon, whilst it's dark (sun below -6degrees) ?

**Comment by cosinekitty** [comment]
Hi @marcmayo and welcome to the project! I can help you with this.

My approach for this would be to use the [DefineStar](https://github.com/cosinekitty/astronomy/tree/master/source/js#definestarbody-ra-dec-distancelightyears) function to define the star you are interested in as a custom body. Then you can pass this body to [SearchRiseSet](https://github.com/cosinekitty/astronomy/tree/master/source/js#SearchRiseSet) to find the next time it rises. When you find the rise time, use that same time to find the Sun's altitude angle via the Horizon function, as demonstrated at [this line](https://github.com/cosinekitty/astronomy/blob/e3fc95a2fbe3a97621a460ad24e596e2528e3144/demo/nodejs/positions.js#L62) of the positions.js demo. If the Sun's altitude is still above &minus;6&deg;, you can use the AddDays function to add, say, 0.1 days to the rise time, and call SearchRiseSet again in a loop. Keep looping until you find the dark enough time.

I don't know which of the programming languages you are interested in, so I gave JavaScript links, but a similar  approach will work for the C, C#, Python, or Kotlin versions.

I hope this helps, and let us know if you run into problems, or that it works fine as is.
